### PR TITLE
feat: have all config via cli and have default provide for picocli

### PR DESCRIPTION
**Pull request by maxandersen**
I got tired of having to edit the config to try things out so changed
config reading to be default provider for picocli so config will provide
defaults but you can override with command line arguments.

### CODE REVIEW

**Comment by fahad-israr** [comment]
This is quite a good thing. In future we can have this thing integrated with the browser extension as well. (i.e The user can chose to open with the existing configurations in config file or can chose a one time configuration  to launch starfix)

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 1]
```diff
@@ -55,31 +62,13 @@ public int cloneCmd(@Parameters(index = "0") String url) {
...
-
-            Config config = mapper.readValue(configFile, Config.class);
-
-            ide = config.ide;
-            clone_path = config.clone_path;
-            if (ide == null || clone_path == null)
-                editConfig(); // Incase of absence of configuration in file Launch Config
+        if(ide==null) {
+            throw new IllegalArgumentException("Editor not specifed, please specify via --editor or use config to set it up.");
+        }
```
Comment (fahad-israr) [inline cli/src/main/java/dev/starfix/Starfix.java:68]:
```suggestion
        new CommandLine(new Starfix()).usage(System.out); // Will invoke Picocli Help
         }
```

**Comment by fahad-israr** [comment]
- So now if we dont have the config file(when the user runs starfix first time), he'll get something like this
```java.lang.IllegalArgumentException: Clone path not specifed, please specify via --clone-path or use config to set it up.```  along with other stacktrace like :
 ``` at dev.starfix.Starfix.cloneCmd(Starfix.java:71)
	at dev.starfix.Starfix_Subclass.cloneCmd$$superforward1(Starfix_Subclass.zig:142)
	at dev.starfix.Starfix_Subclass$$function$$4.apply(Starfix_Subclass$$function$$4.zig:33)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
```

- What I was thinking is that shouldn't it display picocli help instead ? 
```  new CommandLine(new Starfix()).usage(System.out); // Will invoke Picocli Help ```

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 2]
```diff
@@ -55,31 +62,13 @@ public int cloneCmd(@Parameters(index = "0") String url) {
...
 
-        } catch (Exception e) {
-            e.printStackTrace();
+        if(clone_path==null) {
+            throw new IllegalArgumentException("Clone path not specifed, please specify via --clone-path or use config to set it up.");
         }
```
Comment (fahad-israr) [inline cli/src/main/java/dev/starfix/Starfix.java:72]:
```suggestion
         new CommandLine(new Starfix()).usage(System.out); // Will invoke Picocli Help
        }
```

Comment (maxandersen) [reply cli/src/main/java/dev/starfix/Starfix.java:72]: not following this suggested change - printing the long usage help when we know exactly what the user needs seems not good ? do please open sepaerate PR if you think there is still a need.

**Pull request merged**